shared/actions/favorite.js

```diff
@@ -262,11 +267,33 @@ function _notify (state) {
   previousNotifyState = newNotifyState
 }
 
+function * _setupKBFSChangedHandler (): SagaGenerator<any, any> {
+  yield put((dispatch: Dispatch) => {
+    const debouncedKBFSStopped = _.debounce(() => {
+      const badgeAction: Action = badgeApp('kbfsUploading', false)
+      dispatch(badgeAction)
+      dispatch({type: Constants.kbfsStatusUpdated, payload: {isAsyncWriteHappening: false}})
+    }, 2000)
+
+    engine().setIncomingHandler('keybase.1.NotifyFS.FSSyncActivity', ({status}) => {
```

Comment:
This might be a common pattern we'll use, so maybe next time we should clean up this pattern